### PR TITLE
[APPS-531] Create video sink

### PR DESCRIPTION
**Pull request by katelyn-gallagher**
[//]: # (Pull request descriptions are to help other engineers and QA review your pull request)
[//]: # (It's often useful to explain "why" you've chosen this approach or "how" you've accomplished something if it's too verbose to be captured in comments)

## Changes

[//]: # (List of changes, organized by JIRA ticket)
[APPS-531](https://blinemedical.atlassian.net/browse/APPS-531)
- Create a video sink, and attach that to the video track
- Create `VideoAudioFileRenderer` for the sink
- Pipe along the frame rate info that's specified for the videoTrack

## Testing

[//]: # (Steps for testing the changes)
- As long as it builds, we're happy. This isn't wired up yet.

## Out of scope

[//]: # (Known issues which are not addressed in this PR)
- N/A

[APPS-531]: https://blinemedical.atlassian.net/browse/APPS-531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/src/main/java/com/oney/WebRTCModule/MediaRecorderImpl.java [hunk 1]
```diff
@@ -34,10 +42,16 @@ protected void start(File file) throws Exception {
 
         // Make sure videoTrack exists before adding sink
         if (videoTrack != null) {
-            // Create a new VideoAudioFileRenderer using the file path and EglUtils.getRootEglBaseContext()
-            // VideoAudioFileRenderer will write to the file
-            // Add sink to videoTrack (https://chromium.googlesource.com/external/webrtc/+/HEAD/sdk/android/api/org/webrtc/VideoTrack.java#31)
-            // Log.i(TAG, "Started media recorder! " + videoAudioFileRenderer.toString());
+            // Make sure directories exists before writing a file
+            file.getParentFile().mkdirs();
```
Comment (katelyn-gallagher) [inline android/src/main/java/com/oney/WebRTCModule/MediaRecorderImpl.java:46]: This is the only safety check I've added for file safety. I could definitely be more thorough than this if you'd like. I also kinda hate how they did this, but I know it's stolen straight from flutter so it must be effective.

File: android/src/main/java/com/oney/WebRTCModule/VideoAudioFileRenderer.java
Comment (katelyn-gallagher) [inline android/src/main/java/com/oney/WebRTCModule/VideoAudioFileRenderer.java:1]: This is almost identical to your version in your example branch. The only differences are:
- I pulled out everything related to audio
- I piped in a realistic frame rate

File: android/src/main/java/com/oney/WebRTCModule/MediaRecorderImpl.java (hunk 1 above)
Comment (Gandri) [reply android/src/main/java/com/oney/WebRTCModule/MediaRecorderImpl.java:46]: Honestly this is up to how much responsibility we want to put on the caller. Also note that in theory whatever we decide impacts the ios as to not make one magically handle folder creations for you and the other doesn't. I think in most cases it's fine for us to do this especially if it doesn't just throw an error if the directories already exist then it's a nicety that's honestly fine.

File: android/src/main/java/com/oney/WebRTCModule/VideoAudioFileRenderer.java
Comment (Gandri) [reply android/src/main/java/com/oney/WebRTCModule/VideoAudioFileRenderer.java:1]: That's fine. I would have asked at some point to have us pipe the framerate and any other relevant data we set when initializing the stream as that just makes sense here

**Review by Gandri — APPROVED** [review]

**Pull request merged**